# Group Project
## Individual assignment 1: data and question

In [2]:
library(readr)

In [5]:
data <- read_csv("HR-Employee-Attrition.csv")
options(width = 36)
head(data)

Rows: 1470 Columns: 35
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (9): Attrition, BusinessTravel, Department, EducationField, Gender, Job...
dbl (26): Age, DailyRate, DistanceFromHome, Education, EmployeeCount, Employ...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Age Attrition BusinessTravel   
1 41  Yes       Travel_Rarely    
2 49  No        Travel_Frequently
3 37  Yes       Travel_Rarely    
4 33  No        Travel_Frequently
5 27  No        Travel_Rarely    
6 32  No        Travel_Frequently
  DailyRate Department            
1 1102      Sales                 
2  279      Research & Development
3 1373      Research & Development
4 1392      Research & Development
5  591      Research & Development
6 1005      Research & Development
  DistanceFromHome Education
1 1                2        
2 8                1        
3 2                2        
4 3                4        
5 2                1        
6 2                2        
  EducationField EmployeeCount
1 Life Sciences  1            
2 Life Sciences  1            
3 Other          1            
4 Life Sciences  1            
5 Medical        1            
6 Life Sciences  1            
  EmployeeNumber ⋯
1 1              ⋯
2 2              ⋯
3 4              ⋯
4 5              ⋯
5 7              ⋯
6 8              ⋯
  RelationshipSatisfaction
1 1                       
2 4                       
3 2                       
4 3                       
5 4                       
6 3                       
  StandardHours StockOptionLevel
1 80            0               
2 80            1               
3 80            0               
4 80            0               
5 80            1               
6 80            0               
  TotalWorkingYears
1  8               
2 10               
3  7               
4  8               
5  6               
6  8               
  TrainingTimesLastYear
1 0                    
2 3                    
3 3                    
4 3                    
5 3                    
6 2                    
  WorkLifeBalance YearsAtCompany
1 1                6            
2 3               10            
3 3                0            
4 3                8            
5 3                2            
6 2                7            
  YearsInCurrentRole
1 4                 
2 7                 
3 0                 
4 7                 
5 2                 
6 7                 
  YearsSinceLastPromotion
1 0                      
2 1                      
3 0                      
4 3                      
5 2                      
6 3                      
  YearsWithCurrManager
1 5                   
2 7                   
3 0                   
4 0                   
5 2                   
6 6

The dataset contains the information about erevy employees, such as their personal information and information about their status in the company. The focus of dataset is the attrition status and can be used to help HR to decide to retain this employee or not. There are 1470 observations in the dataset and 35 variables. Below is the description of dataset。

|Variable |Type |Description |
|---------|-----|------------|
|Age      |continuous  |Age of Employee|
|Attrition| categorical|If the employee has been quited or some other reason|
|BusinessTravel| categorical| Rate of business travel|
|DailyRate| continuous| Amount of pay per day|
|Department|categorical| Department of employee|
|DistanceFromHome| continuous| Distance from home to work in miles|
|Education|categorical| Education level|
|EducationField| categorical| Field of study|
|EmployeeCount| continuous| Count of employee|
|EmployeeNumber| categorical | Employee number|
|EnvironmentSatisfaction|categorical|Employee satisfaction(from 1 to 4) at work|
|Gender| categorical | gender of employee|
|HourlyRate|continuous|employee hourly rate|
|JobInvolvement| categorical| Level of involvment|
|JobLevel|categorical|Level of senority|
|JobRole| categorical|Employee's role in the company|
|JobSatisfaction|categorical|Satisfaction of employee|
|MaritalStatus|categorical|Marital Status of employee|
|MonthlyIncome|continuous|Monthly income of an employee|
|MonthlyRate|continuous|Monthly rate of an employee|
|NumCompaniesWorked|continuous|Number of companies that employee has worked at before|
|Over18|categorical|Employee is over 18 or not|
|OverTime|categorical|Whether the employee works overtime|
|PercentSalaryHike|continuous|Percent of salary hike|
|PerformanceRating|categorical|performance rate|
|RelationshipSatisfaction|categorical|satisfaction of relationship |
|StandardHours|continuous| per week standard work hours|
|StockOptionLevel|categorical| company stock option level|
|TotalWorkingYears| continuous| total working years|
|TrainingTimesLastYear|continuous|training time|
|WorkLifeBalance|categorical|balance between work and life|
|YearAtCompany|continuous|total years at current company|
|YearsInCurrentRole| continuous| total years in current role|
|YearsSinceLastPromotion|continuous|number of years since last promotion|
|YearsWithCurrManager| continuous|Years working under the current manager|

Question: How is attrition (as our response variable) affected by the variables in the dataset. For the explanatory variables, we will discuss it more in the next several phases of the project.